### ISSUE TITLE
Import existing layouts used by setxkbmap

### ISSUE THREAD

**Issue by tobiasBora** [rvaiya/keyd#755]
Thanks for this cool tool. Some layouts may be very long to encode in keyd (e.g. the bépo afnor layout with latin/greek/math/… support, or qwerty international). These files should not be too hard to parse, at least as a first approximation, as they look like:
```
key <AC05> { type[group1] = "FOUR_LEVEL", [ comma, semicolon, rightsinglequotemark, dead_belowcomma ] }; // , ; ’ ,
```

**Comment by nsbgn** [comment]
The problem is that `setxkbmap` does not deal only with the low-level keycodes, but also with how they are interpreted in X11, which keyd necessarily knows nothing about. It's probably possible and may be useful to write a little script to translate xkb layouts into keyd configs, using compose sequences to get the missing symbols, but it would be fickle.

**Comment by tobiasBora** [comment]
I see thanks. Guess it would still be a first step to gain some time.

**Comment by rvaiya** [comment]
> It's probably possible and may be useful to write a little script to translate xkb layouts into keyd configs, assuming an en-US layout and using compose sequences to get any missing symbols, but it would be fickle.

I used to have an awk script lying around which I used to generate some of the keyd layouts from xkb files. Writing a proper XKB parser is a PITA, so it didn't cover most corner cases.

I can't seem to find the original script, so here is my attempt to recreate it in python:

``` 
#!/usr/bin/env python3

import re
import sys
from collections import defaultdict

xkbToKeydcode = {
    'TLDE':	'grave',
    'AE01':	'1', 'AE02':	'2', 'AE03':	'3', 'AE04':	'4', 'AE05':	'5', 'AE06':	'6',
    'AE07':	'7', 'AE08':	'8', 'AE09':	'9', 'AE10':	'0', 'AE11':	'minus', 'AE12':	'equal',
    'AD01':	'q', 'AD02':	'w', 'AD03':	'e', 'AD04':	'r', 'AD05':	't', 'AD06':	'y',
    'AD07':	'u', 'AD08':	'i', 'AD09':	'o', 'AD10':	'p', 'AD11':	'bracketleft', 'AD12':	'bracketright',
    'AC01':	'a', 'AC02':	's', 'AC03':	'd', 'AC04':	'f', 'AC05':	'g', 'AC06':	'h',
    'AC07':	'j', 'AC08':	'k', 'AC09':	'l', 'AC10':	'semicolon', 'AC11':	'apostrophe', 'BKSL':	'backslash',
    'AB01':	'z', 'AB02':	'x', 'AB03':	'c', 'AB04':	'v', 'AB05':	'b', 'AB06':	'n',
    'AB07':	'm', 'AB08':	'comma', 'AB09':	'period', 'AB10':	'slash'
}

layers = defaultdict(lambda: {})

for line in sys.stdin:
    try:
        m = re.match('\s*key <([^>]*)>.*// *(.*)', line)
        key = xkbToKeydcode[m[1]]
        
        for (i, symbol) in enumerate(m[2].split(' ')):
            layers[i][key] = symbol
    except:
        pass

print(f'[main]')
for (k,v) in layers[0].items():
    print(f'{k} = {v}')
print()

if 1 in layers:
    print(f'[shift]')
    for (k,v) in layers[1].items():
        print(f'{k} = {v}')
print()

if 2 in layers:
    print(f'[altgr]')
    for (k,v) in layers[2].items():
        print(f'{k} = {v}')
```


It relies on the xkb definition containing the actual unicode symbols as a comment (which many of them seem to do).

A more robust approach would be to scrape symbol/name mappings from the Compose file, but then you start
going down the XKB configuration rabbit hole and never emerge :P.
